documentation/src/main/asciidoc/userguide/chapters/statistics/Statistics.adoc

```diff
@@ -0,0 +1,159 @@
+
+Hibernate can gather all sorts of statistics which can help you get a better insight into what Hibernate does behind the scenes.
+
+By default, the statistics are not collected because this incurs an additional processing and memory overhead. To instruct Hibernate to start collecting statistics, you need to set the `hibernate.generate_statistics` configuration property to `true`:
+
+====
+[source,xml]
+----
+<property
+    name="hibernate.generate_statistics"
+    value="true"
+/>
+----
+====
+
+[[statistics-methods]]
+=== org.hibernate.stat.Statistics methods
+
+The Hibernate statistics are made available via the
+https://docs.jboss.org/hibernate/orm/{majorMinorVersion}/javadocs/org/hibernate/stat/Statistics.html[`Statistics`] interface which exposes the following methods:
+
+[[statistics-general]]
+==== General statistics methods
+
+`isStatisticsEnabled`:: Are statistics enabled?
+`setStatisticsEnabled(boolean b)`:: Enable statistics based on the provided parameter.
+`clear`:: Reset all statistics.
+`logSummary`:: Print a summary of the current statistics into the application log.
+`getStartTime`:: The milliseconds (JVM standard `currentTimeMillis()`) from which all statistics accessed since the initial creation of this Statistics instance or the last `clear()` method call.
+
+
+[[statistics-aggregates]]
+==== Aggregated statistics methods
+
+`getQueries`:: Get executed query strings. The maximum number of queries tracked by the Hibernate statistics is given by the `hibernate.statistics.query_max_size` property.
+`getEntityStatistics(String entityName)`:: Find entity statistics for the given name.
+`getCollectionStatistics(String role)`:: Get collection statistics per role (collection name).
+`getNaturalIdStatistics(String entityName)`:: Get the Hibernate-specific natural id resolution statistics for the given entity.
+`getQueryStatistics(String queryString)`:: Get the statistics for the given query string (JPQL/HQL or native SQL).
+`getDomainDataRegionStatistics(String regionName)`:: Get the second-level cache statistics per domain data (entity, collection, natural-id) region.
+`getQueryRegionStatistics(String regionName)`:: Get the second-level cache statistics per query region.
+`getCacheRegionStatistics(String regionName)`:: Get statistics for either a domain-data or query-result region
+(this method checks both, preferring domain data region if one).
+
+[[statistics-session-factory]]
+==== SessionFactory statistics methods
+
+`getEntityNames`:: Get the names of all entities configured with the current `SessionFactory`.
+`getCollectionRoleNames`:: Get the names of all collection roles configured with the current `SessionFactory`.
+
+[[statistics-session]]
+==== Session statistics methods
+
+`getSessionCloseCount`:: Global number of sessions that got closed.
+`getSessionOpenCount`:: Global number of sessions that got opened.
+`getFlushCount`:: Get the global number of flush executed by sessions (either manual or automatic).
```

Comment:
`flushes` or `flush operations`?
`by session` without s, I think

Reply:
Changed to:

> Get the global number of flush operations executed (either manual or automatic).